storm-core/src/jvm/org/apache/storm/utils/TopologySpoutLag.java

```diff
@@ -43,26 +43,44 @@
     public static List<Map<String, Object>> lag (StormTopology stormTopology, Map topologyConf) {
         List<Map<String, Object>> result = new ArrayList<>();
         Map<String, SpoutSpec> spouts = stormTopology.get_spouts();
-        Object object = null;
+        String className = null;
         for (Map.Entry<String, SpoutSpec> spout: spouts.entrySet()) {
             try {
                 SpoutSpec spoutSpec = spout.getValue();
                 ComponentObject componentObject = spoutSpec.get_spout_object();
-                object = Utils.getSetComponentObject(componentObject);
```

Comment:
The problem starts here. For many situation we can't deserialize Spout object since UI doesn't have dependency for that Spout, for example, storm-kafka or storm-kafka-client.